docs/specs/meta-clients/web3modal/api.md

```diff
@@ -1,142 +1,248 @@
+    }
+  ]
+}
 ```
 
-#### Known Static Asset IDs
+### `GET /v2/wallet-image/{size}/{wallet_id}`
+
+Returns the image logo for the specified wallet ID.
+
+#### Route Parameters
+
+- `wallet_id`: The unique identifier for the wallet
+- `size`: The predefined image size, described as `{width}x{height}` (see [Known Image Sizes](#known-image-sizes))
+
+#### Response
+
+- A binary image file (Blob)
+
+### `GET /v2/asset-image/network/{size}/{asset_id}`
+
+Fetches image for public assets like network logos using well-defined CAIP identifiers.
+
+#### Route Parameters
+
+- `asset_id`: CAIP identifier for the network (e.g., `eip155-1`)
+- `size`: The predefined image size, described as `{width}x{height}` (see [Known Image Sizes](#known-image-sizes))
+
+#### Example API Requests
+
+- Ethereum: `GET /v2/asset-image/network/56x56/eip155-1`
+- Arbitrum: `GET /v2/asset-image/network/56x56/eip155-42161`
+- Avalanche: `GET /v2/asset-image/network/56x56/eip155-43114`
+- Binance Smart Chain: `GET /v2/asset-image/network/56x56/eip155-56`
+- Fantom: `GET /v2/asset-image/network/56x56/eip155-250`
+- Optimism: `GET /v2/asset-image/network/56x56/eip155-10`
+- Polygon: `GET /v2/asset-image/network/56x56/eip155-137`
+- Gnosis: `GET /v2/asset-image/network/56x56/eip155-100`
+- Evmos: `GET /v2/asset-image/network/56x56/eip155-9001`
+- ZkSync: `GET /v2/asset-image/network/56x56/eip155-324`
+- Filecoin: `GET /v2/asset-image/network/56x56/fil-314`
+- IoTeX: `GET /v2/asset-image/network/56x56/eip155-4689`
+- Metis: `GET /v2/asset-image/network/56x56/eip155-1088`
+- Moonbeam: `GET /v2/asset-image/network/56x56/eip155-1284`
+- Moonriver: `GET /v2/asset-image/network/56x56/eip155-1285`
+- Zora: `GET /v2/asset-image/network/56x56/eip155-7777777`
+- Celo: `GET /v2/asset-image/network/56x56/eip155-42220`
+- Base: `GET /v2/asset-image/network/56x56/eip155-8453`
+- Aurora: `GET /v2/asset-image/network/56x56/eip155-1313161554`
+
+#### Response
+
+- A binary image file (Blob)
 
-There are a number of known static asset IDs that can be used to fetch images, e.g. for networks and connectors:
+### `GET /v2/asset-image/connector/{size}/{connector_id}`
+
+Fetches image for connector logos using descriptive names.
+
+#### Route Parameters
+
+- `connector_id`: Descriptive name for the connector (e.g., `coinbase-wallet`)
+- `size`: The predefined image size, described as `{width}x{height}` (see [Known Image Sizes](#known-image-sizes))
+
+#### Example API Requests
+
+- Coinbase Wallet: `GET /v2/asset-image/connector/56x56/coinbase-wallet`
+- Safe: `GET /v2/asset-image/connector/56x56/safe`
+- Ledger: `GET /v2/asset-image/connector/56x56/ledger`
+- WalletConnect: `GET /v2/asset-image/connector/56x56/wallet-connect`
+- Injected (Generic Web3 Injected Connectors): `GET /v2/asset-image/connector/56x56/injected`
+
+#### Response
+
+- A binary image file (Blob)
+
+### `GET /v2/asset-image/token/{size}/{token_symbol}`
+
+Fetches image for known tokens using the token's ticker symbol.
```

Comment:
Isn't it possible for ticket symbols to overlap? Maybe there is a more robust way to do this? E.g. chain ID + token ID? How is swaps doing this right now?

Reply:
I agree we at least need to scope this by network given ticker might repeat (eg OP ETH vs Mainnet ETH) 

`.../token/{size}/{caip-2}/{tokenAddress}` would be the most exact, `.../token/{size}/{caip-2}/{tokenAddress}` seems more similar to current structure

Reply:
Good points 👍 

One question: will requiring `tokenAddress` for querying cause more lookup requirements on the SDK side (which we want to explicitly avoid with these changes) or can you query `/{caip-2}/{tokenAddress}` a priori @tomiir?

How is this being queried in the clients currently with the existing `/public/getTokenImage/ETH` approach?

Reply:
@Cali93 was just checking for implementations. 

This was originally intended for token images from Onramp activity- The address is not avail on the response from the tx status https://docs.cdp.coinbase.com/onramp/docs/api-reporting/#transaction-status

but checked the API specs and the response from the options returns the contract address already, so we should have that information available if we set it up correctly

https://docs.cdp.coinbase.com/onramp/docs/api-configurations/#example-requestresponse-1

Reply:
I was mainly considering a portfolio view; looks like the balances endpoint [returns the contract address](https://specs.walletconnect.com/2.0/specs/servers/blockchain/blockchain-api#success-response-body-5).

Reply:
For onramp API, I would propose that Blockchain API injects the contract address, abstracting away potential conflicts. For now this could be a hardcoded lookup table in Blockchain API

Reply:
There is also the consideration of cache-busting. We have long/immutable-cached responses. For `/v2/wallets` we could change the wallet ID when the image is updated. But for others this is more tricky such as network images.

Reply:
I propose we use [CAIP-19](https://github.com/ChainAgnostic/CAIPs/blob/main/CAIPs%2Fcaip-19.md) for this endpoint. So it would look something like this:

```
GET /v2/asset-image/token/64x64/eip155%3A1%2Ferc20%3A0x6b175474e89094c44da98b954eedeac495271d0f`
```

[eip155 namespace for CAIP-19](https://namespaces.chainagnostic.org/eip155/caip19)

Reply:
Nice, didn't know of CAIP19. It seems like the most explicit given we get the contract address from the API @chris13524

Reply:
Nice one on the CAIP-19 suggestion @chris13524 💯 🙏 

One technicality we need to resolve, without explicit URL-encoding the DAI example is:
```
eip155:1/erc20:0x6b175474e89094c44da98b954eedeac495271d0f
```

The slash will be interpreted as an additional path segment, and we can't expect clients to do partial URL-encoding, so we either:
* replace the `/` with another unambiguous value
* Treat the values before and after the `/` like real path segments, i.e. `GET /v2/asset-image/token/{chain_id}/{asset_namespace + ":" + asset_reference}`

Personally I'd prefer to just replace with e.g. an underscore (`eip155:1_erc20:0x6b175474e89094c44da98b954eedeac495271d0f`) because using part of the identifier as a path seems odd.

Reply:
> we can't expect clients to do partial URL-encoding

@bkrem what do you mean here? URL encoding is just as easy as base64 encoding. For example in JS with a function such as `encodeURIComponent()`:

```
`GET /v2/asset-image/token/64x64/${encodeURIComponent(caip19Id)}`
```

Reply:
> URL encoding is just as easy as base64 encoding. For example in JS with a function such as encodeURIComponent()

Sure I get the approach, but I think it's odd to expect that kind of accounting work from a client. Seems like this already gets awkward when you want to query this endpoint with e.g. `curl`.